src/functions/assignment_execution_handler/index.py

```diff
@@ -45,20 +45,48 @@ class UnknownException(Exception):
 logger = error_handler.get_logger()
 
 
+# Is Identity Center delegated admin?
+use_delegated_admin = None
+
+# Get SSO Instance
+sso_delegated_admin = None
+
 # Get SSO Instance
+management_account_id = os.getenv("MANAGEMENT_ACCOUNT_ID", "012345678901")
 sso_admin_role_arn = os.getenv(
     "SSO_ADMIN_ROLE_ARN",
     "arn:aws:iam::112223334444:role/assignment-management-role",
 )
-assumed_role_session = assume_role(session, sso_admin_role_arn)
-sso = None
+assumed_admin_role_session = assume_role(session, sso_admin_role_arn)
+sso_admin = None
 
 
 def handler(event, context):
     # TODO make proper call outside handler work with tests
-    global sso
-    if sso == None:
-        sso = SsoService(assumed_role_session)
+    global sso_admin
+    global sso_delegated_admin
+    global use_delegated_admin
+
+    # check if delegated admin is enabled
+    if use_delegated_admin is None:
```

Comment:
please use black to format this file. it should pickup the settings from pyproject.toml

Reply:
I sued the black formatter for the file